### PR TITLE
Improve packages webui

### LINKED ISSUE #116: Members Link in SynoCommunity Website Footer Is Bad
### Is this a new Bug?

- [X] I checked that the bug hasn't been reported before

### What happened?

The link "Maintained by [SynoCommunity](https://github.com/SynoCommunity?tab=members)" at the bottom of https://synocommunity.com/ , in the footer is bad.

Should point to https://github.com/orgs/SynoCommunity/people

Also, all the "View details links under "Easy Install", "Bleeding Edge", and "Open Source" just link to itself and seem pointless, gives no further details, recommend removing the links or truly give more info.

### Reproduction steps

n/a

### PR DESCRIPTION
**Pull request by hgy59**
###  fix beta label and use different labels for active and inactive packages

- apply "beta" to package versions instead of packages
- use green labels for active packages
- use default (gray) label for inactive packages and add tooltip telling the package is inactive
- update version date (cut microseconds)
- fix unclosed span tag in package.html

###  show packages without active versions (leftover of #106)

- do not hide package in packages when there is no active version (https://github.com/SynoCommunity/spksrc/issues/5803)
- ~remove obsolete TODOs~

### fix webpage footer (fixes #116)
- correct link for SynoCommunity members

### CODE REVIEW

**Comment by hgy59** [comment]
@mreid-tt this includes a proposal to distinguish active an inactive packages.

It is still a static solution and does not add a user filter to include/exclude inactive packages.

**Comment by mreid-tt** [comment]
@hgy59, great work! Pulled into my test environment and all the improvements seem to be there as described. 🚀

**Comment by hgy59** [comment]
> I've not interrogated the modifications you did to the tests but I will be guided by your validation.

packages frontend test changes:
- beta labels are not shown anymore
- packages with only inactive versions are shown now

package (singular) frontend test changes:
- beta label tests added (according to former packages tests)

File: spkrepo/tests/test_frontend.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def test_get_logged_user(self):
 
 
 class PackagesTestCase(BaseTestCase):
-    def test_get_active_stable(self):
+    def test_get_active(self):
         build = BuildFactory(version__report_url=None, active=True)
```
Comment (mreid-tt) [inline spkrepo/tests/test_frontend.py:28]: If this is now supposed to be a mirror image of `test_get_not_active`, then perhaps the build factories should be identical by adding or removing this parameter:

```
version__report_url=None
```

File: spkrepo/tests/test_frontend.py [hunk 2]
```diff
@@ -35,45 +35,67 @@ def test_get_active_stable(self):
         )
         self.assertNotIn("beta", response.data.decode())
```
Comment (mreid-tt) [inline spkrepo/tests/test_frontend.py:37]: If we aren't testing for beta in this test, should this assertion be removed?

**Comment by hgy59** [comment]
This are the updated views:

beta on verstion instead of package (revisions 10 and 11 are beta, rev 9 not):
spkrepo_beta


inactive package with tooltip (ppc824x and ppc854x of rev 3 are inactive), active packages have green label now (screen shot without mouse cursor):
spkrepo_inactive_tooltip

**Comment by mreid-tt** [comment]
> This are the updated views:

I was thinking more about the tests. Is there a way for it to test that the builds that are shown in the package webpage have the active/inactive colours you added? I don't recall seeing that in your new tests.

**Comment by mreid-tt** [comment]
Hello @hgy59, I encountered a situation while working on the admin side that I believe you can address on the frontend and incorporate into this pull request. The issue arises when a package is created without any versions, and upon manually navigating to the package URL, the server returns an internal server error. I propose the following code modification to handle this scenario:

https://github.com/SynoCommunity/spkrepo/blob/0ce2d0235fccb62df07d23dfe1898b9c7606b974/spkrepo/views/frontend.py#L100

Change to:

```
    if package is None or not package.versions:
```

**Comment by mreid-tt** [comment]
Hi @hgy59, as another suggestion I would remove the duplicate password reset link on the login page by deleting this line:

https://github.com/SynoCommunity/spkrepo/blob/0ce2d0235fccb62df07d23dfe1898b9c7606b974/spkrepo/templates/security/login_user.html#L6

**Comment by mreid-tt** [comment]
@hgy59, I'd love to hear your thoughts on my suggestions. Alternatively, if you prefer, I can make the amendments directly through a commit for you.

File: spkrepo/tests/test_frontend.py (hunk 1 above)
Comment (mreid-tt) [reply spkrepo/tests/test_frontend.py]: @hgy59, I'd love to hear your thoughts on these review comments as well.

Comment (hgy59) [reply spkrepo/tests/test_frontend.py]: A report url != None means that it is a beta version.
Those tests document that there is not beta label shown anymore on the packages page.

Comment (mreid-tt) [reply spkrepo/tests/test_frontend.py]: Okay. In that case should you also add it to `test_get_not_active(self)`:
https://github.com/SynoCommunity/spkrepo/blob/fed32c9e347c8b706cc16481b7eca3c9f6995d4a/spkrepo/tests/test_frontend.py#L39

File: spkrepo/tests/test_frontend.py (hunk 2 above)
Comment (mreid-tt) [reply spkrepo/tests/test_frontend.py]: @hgy59, I'd love to hear your thoughts on this review comment as well.

File: spkrepo/tests/test_frontend.py (hunk 1 above)
Comment (hgy59) [reply spkrepo/tests/test_frontend.py]: there is no `test_get_not_active` anymore.

Comment (mreid-tt) [reply spkrepo/tests/test_frontend.py]: Sorry, I was comparing these two:

https://github.com/SynoCommunity/spkrepo/blob/5bce17219319e29dc4b2648b64ea706c28e49a45/spkrepo/tests/test_frontend.py#L30-L31

https://github.com/SynoCommunity/spkrepo/blob/5bce17219319e29dc4b2648b64ea706c28e49a45/spkrepo/tests/test_frontend.py#L43-L44

Comment (hgy59) [reply spkrepo/tests/test_frontend.py]: @mreid-tt thanks, I think now I've got it 😃

**Comment by mreid-tt** [comment]
@hgy59, I hope you don't mind but I reverted a change to `spkrepo/views/admin.py` as this was addressed in my PR #112. This will prevent a merge conflict between the two.

**Comment by mreid-tt** [comment]
@Diaoul, @publicarray, do either of you have time to give this a review and approval? Eager to get these into production.

**Comment by mreid-tt** [comment]
I was looking at issue #116 that was raised regarding the webpage footer. In looking at the code and the website they do not seem to be in sync. For example the footer in the source code is:

https://github.com/SynoCommunity/spkrepo/blob/021528dbce103075b34ad7997eef914693826743/spkrepo/templates/layout.html#L91-L97

However on the website the footer has this code:

```
    <footer class="footer" role="contentinfo">
      <div class="container">
        <p>Designed by Antoine Bertin.</p>
        <p>Maintained by <a href="https://github.com/SynoCommunity?tab=members">SynoCommunity</a> with the help of <a href="https://github.com/SynoCommunity/spksrc/graphs/contributors">contributors</a>.</p>
        <p>Code licensed under <a href="https://github.com/SynoCommunity/spkrepo/blob/master/LICENSE" target="_blank">MIT</a>.</p>
      </div>
    </footer>
```

I would have normally proposed a fix to include in this PR but the codebase and the website don't seem to align. Any thoughts on this? @publicarray @Diaoul ?

**Comment by Diaoul** [comment]
Yes some templates are overridden locally on the server. The only way to change them is to SSH to it.

**Comment by mreid-tt** [comment]
> Yes some templates are overridden locally on the server. The only way to change them is to SSH to it.

Hmm, any particular reason for this manual editing? Should I move the server-side HTML into the codebase so they can be in sync?

**Comment by mreid-tt** [comment]
@hgy59 given the comment above from @Diaoul, I've brought the code in from the live website into the codebase so they would be in sync and resolved the broken members link raised in #116.

**Comment by ymartin59** [comment]
I wonder if @Diaoul expects to include synocommunity.com specific templates into code base, as spkrepo may be deployed by other users.

**Comment by mreid-tt** [comment]
> I wonder if @Diaoul expects to include synocommunity.com specific templates into code base, as spkrepo may be deployed by other users.

As far as I can observe the code is very explicit to SynoCommunity in the templates. My suspicion is that there was a minor HTML change in the footer that was needed sometime ago and the change was made as a direct edit on the server and never brought back into the codebase (I've seen a similar thing in the database schema). This last commit was meant to bring them back in sync so that we don't have this to worry about it moving forward.

To check for any others (that we can't see without server access), according to [this blog](https://www.createit.com/blog/compare-git-and-server-files/) we can run these commands on the server:

_To display file names that are NOT under version control:_
```
git clean -fdxn
```

_To display all changes in the GIT files just use:_
```
git diff .
```

So if there are any other files which are different we can sync them up. This makes the server easier to maintain and if we ever had to move to another server there would be less non-standard variables to worry about.

**Comment by publicarray** [comment]
@mreid-tt looks good to me, with this the spkrepo is going to be more specific for our uses and no longer as generic. 

The reason for removing the password reset I assume is because we can't send emails? I would have thought it's best to leave it in regardless so that at least a reset code can be generated, even if it isn't sent.

**Comment by publicarray** [comment]
> Hi @hgy59, as another suggestion I would remove the duplicate password reset link on the login page by deleting this line:
> 
> https://github.com/SynoCommunity/spkrepo/blob/0ce2d0235fccb62df07d23dfe1898b9c7606b974/spkrepo/templates/security/login_user.html#L6

Oh duplicate I see

**Pull request merged**